cmd/nsm-sriov-manager/nsm-sriov-manager.go

```diff
@@ -0,0 +1,418 @@
+// Copyright (c) 2018 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	netDirectory    = "/sys/class/net/"
+	pciDirectory    = "/sys/bus/pci/devices"
+	sriovCapable    = "/sriov_totalvfs"
+	sriovConfigured = "/sriov_numvfs"
+	// nsmSRIOVDefaultNetworkServiceName defines a default network service name for all
+	// SRIOV VFs found on a host. Further editing might be required to map VF to real
+	// Network Services
+	nsmSRIOVDefaultNetworkServiceName = "networkservicemesh.io/sriov/nsm-default"
+)
+
+// VF describes a single instance of VF
+type VF struct {
+	NetworkService string `yaml:"networkService" json:"NetworkService"`
+	pciAddr        string // `yaml:"pciAddr" json:"pciAddr"`
+	ParentDevice   string `yaml:"parentDevice" json:"parentDevice"`
+	VFLocalID      int32  `yaml:"vfLocalID" json:"vfLocalID"`
+	pciVendor      string // `yaml:"pciVendor" json:"pciVendor"`
```

Comment:
Won't these be required for scheduling, in which case, shouldn't they have yaml tags?  I don't ask for 'any passthrough device' I ask for 'a network device and one that I have the driver for', so these are interesting when requesting devices and should be in the device map.

Reply:
Also missing, for addressing in future: NUMA attachment

Reply:
Scheduling will be accomplished by `network service name` since it is what we advertise to kubelet. Each `network service` will have a set of vfio devices in a form `/dev/vfio/{iommu_group}`. So when pod requests `network service 1` k8s scheduler will know where it is located and ship pod to the compute node hosting this service.

Please suggest a variable name for NUMA, or just NUMA is sufficient?